File: dvbViewerController/src/main/java/org/dvbviewer/controller/ui/widget/PasswordEditTextPreference.java
package org.dvbviewer.controller.ui.widget;

import android.content.Context;
import android.text.InputType;
import android.util.AttributeSet;

import androidx.preference.EditTextPreference;

public class PasswordEditTextPreference extends EditTextPreference implements InputTypePref {

    public PasswordEditTextPreference(Context context, AttributeSet attrs, int defStyleAttr,
                                      int defStyleRes) {
        super(context, attrs, defStyleAttr, defStyleRes);
    }

    public PasswordEditTextPreference(Context context, AttributeSet attrs, int defStyleAttr) {
        super(context, attrs, defStyleAttr);
    }

    public PasswordEditTextPreference(Context context, AttributeSet attrs) {
        super(context, attrs);
    }

    public PasswordEditTextPreference(Context context) {
        super(context);
    }

    @Override
    public int getInputType() {
        return InputType.TYPE_CLASS_TEXT | InputType.TYPE_TEXT_VARIATION_PASSWORD;
    }

}

File: dvbViewerController/src/main/java/org/dvbviewer/controller/ui/widget/PasswordEditTextPreference.kt
package org.dvbviewer.controller.ui.widget

import android.content.Context
import android.text.InputType
import android.util.AttributeSet

import androidx.preference.EditTextPreference

class PasswordEditTextPreference : EditTextPreference, InputTypePref {

    override val inputType: Int
        get() = InputType.TYPE_CLASS_TEXT or InputType.TYPE_TEXT_VARIATION_PASSWORD

    constructor(context: Context, attrs: AttributeSet, defStyleAttr: Int,
                defStyleRes: Int) : super(context, attrs, defStyleAttr, defStyleRes) {
    }

    constructor(context: Context, attrs: AttributeSet, defStyleAttr: Int) : super(context, attrs, defStyleAttr) {}

    constructor(context: Context, attrs: AttributeSet) : super(context, attrs) {}

    constructor(context: Context) : super(context) {}

}